### ISSUE TITLE
[V6] Autocomplete within Modal broken

### ISSUE THREAD

**Issue by d-a-n42** [mantinedev/mantine#3952]
### What package has an issue

@mantine/core

### Describe the bug

First things first: A big thank you to all maintainers for providing this amazing UI lib 👍 

I just updated from V5 to V6 and all Modals started to break due to overflow issues. Setting `withinPortal` to those elements did not help to fix the issue. I'm using an Autocomplete component with many items. In V5 the use could simply scroll within the list. With V6 the overflow is hidden and is not scollable anymore. I've attached a codesandbox example of the issue. How can I get the Autocomplete list to scroll again w/o having to overwrite the CSS styles?

Basically the expected behaviour is that Autocomplete in a Modal should behave like the Spotlight component.

### What version of @mantine/hooks page do you have in package.json?

6.0.5

### If possible, please include a link to a codesandbox with the reproduced problem

https://codesandbox.io/s/intelligent-breeze-ye2j66?file=/src/App.tsx

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

None

**Comment by rtivital** [comment]
You can set `maxDropdownHeight` – https://codesandbox.io/s/sad-rosalind-51edkb?file=/src/App.tsx

**Comment by rtivital** [comment]
This issue is closed for one of these reasons:

- It was fixed in 7.0
- It is no longer reproducible in 7.0
- It is no longer applicable in 7.0
- It does not have a reproduction

If you think that this issue was closed by mistake, and it is still an issue in version 7.0, please reopen it.
